src/lib/ui/components/Select/Select.svelte

```diff
@@ -89,17 +130,47 @@
     }
   }
 
-  $: onHide = active
-    ? () => {
-        handleToggle();
-        setField(name, $data[name] ?? "");
-      }
-    : undefined;
+  function onHide() {
+    onToggle();
+    onHasValue();
+    setField(name, $data[name] ?? (multiple ? [] : ""));
+  }
+
+  function onRemove(option: SelectOption) {
+    const selection = ($data[name] ?? []) as string[];
+    const newSelection = selection.filter((value) => value !== option.value);
+    setField(name, newSelection);
+  }
+
+  function onClear() {
+    setField(name, multiple ? [] : "");
+    if (!multiple) search = "";
+  }
 
   $: datasets = generateDatas(datas);
-  $: showedValue =
-    options.find(({ value }) => value === $data[name])?.label ||
-    (!!$data[name] ? $data[name] : placeholder);
+
+  $: optionsToShow = (
+    searchable
+      ? options.filter(({ label }) =>
+          label.toLowerCase().includes(search.toLowerCase())
+        )
+      : options
+  ).filter(({ value }) =>
+    multiple
+      ? !(($data[name] ?? []) as string[]).some((v) => v === value)
+      : true
+  );
+
+  $: showedValue = (
+    multiple
+      ? options.filter((option) =>
+          (($data[name] ?? []) as string[]).some(
```

Comment:
🚫 **[eslint]** <[@typescript-eslint/indent](https://typescript-eslint.io/rules/indent)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Expected indentation of 8 spaces but found 10.
```suggestion
        (($data[name] ?? []) as string[]).some(
```